tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/LegacyPopTest.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Identity.Test.Integration.NetFx.HeadlessTests
+{
+    [TestClass]
+    public class LegacyPopTests
+    {
+        [TestMethod]
+        public async Task LegacyPoPAsync()
+        {
+            IConfidentialAppSettings settings = ConfidentialAppSettings.GetSettings(Cloud.Public);
+            X509Certificate2 clientCredsCert = settings.GetCertificate();
+            RsaSecurityKey popKey = CreateRsaSecurityKey();
+
+            var cca = ConfidentialClientApplicationBuilder
+                .Create(settings.ClientId)
+                .WithAuthority(settings.Authority, true)
+                .WithCertificate(clientCredsCert)
+                .WithExperimentalFeatures(true)
+                .WithTestLogging()
+                .Build();
+
+            var result = await cca.AcquireTokenForClient(settings.AppScopes)
+                .WithClientAssertionOverride(
+                    (tokenEndpoint) => CreateMs10ATPOPBodyParams(settings.ClientId, tokenEndpoint, clientCredsCert, popKey, true), 
```

Comment:
The content of `CreateMs10ATPOPBodyParams` is not relevant. But suffice to say it combines the client creds X509 certificate with the POP key and adds it as a param named `request` (and not `client_assertion`)